crates/sparrow-compiler/src/data_context.rs

```diff
@@ -303,6 +303,52 @@ pub struct GroupInfo {
     key_type: DataType,
 }
 
+/// Concurrent file sets allows us to update the file sets from the
+/// python library and see updates reflected within the data context.
+#[derive(Debug, Default, Clone)]
+pub struct ConcurrentFileSets {
```

Comment:
Maybe just call this "FileSets"?

Why do we need multiple file sets, vs adding to the existing file set?

Could this just be `Arc<Mutex<compute_table::FileSet>>`? Doesn't that alreday have a vector inside it?

Reply:
> Why do we need multiple file sets, vs adding to the existing file set?

If and when we do support slicing, each file set would correspond to a specific slice plan

> Could this just be Arc<Mutex<compute_table::FileSet>>? Doesn't that alreday have a vector inside it?

Could, just wanted to abstract a little bit away